Feature/aic 20 notification

luckytzu opened the pull request.

luckytzu committed:
feat(contact): add email sending functionality for user requests

pvcsam reviewed (changes requested):
We said
First : english is the only allowed language
second : ts file not js file
finally: less chatgpt

don't worry. After you resolve it. I have more things to say

+where are the tests and the architecture?

luckytzu committed:
Fix ESLint configuration and Prettier formatting issues

- Updated ESLint config to comply with flat config format
- Defined global variables like 'module', 'require', 'process', and 'console'
- Fixed Prettier formatting issues related to unnecessary spaces

luckytzu committed:
fix(cors): secure CORS configuration for development environment

luckytzu committed:
fix(security): disable X-Powered-By header for improved security

luckytzu committed:
fix(tests): install jest for coverage reporting

luckytzu committed:
fix(tests): correct module path for mailer in server tests

luckytzu committed:
Refactor: Add unit tests for mailer.js and server.js with proper error handling

luckytzu committed:
feat: add mailer.d.ts for type declarations

luckytzu committed:
docs: add comments to test file

luckytzu committed:
chore: convert JS code to TypeScript

ClemZi02 reviewed (changes requested):
don't push file in coverage/

ClemZi02 commented on README.md:
```diff
@@ -4,4 +4,4 @@ Test file location :
 
-Don't forget to delete a.test.ts
\ No newline at end of file
+Don't forget to delete a.test.ts
```
you can delete this line

ClemZi02 commented on eslint.config.mjs:
only devops can modify eslint configuration

ClemZi02 commented on src/server.ts:
```diff
@@ -0,0 +1,64 @@
+import express, { Request, Response } from "express";
+import cors from "cors";
+import { sendEmail } from "./mailer"; // Assurez-vous que mailer.ts est bien typé
```
english comment

ClemZi02 commented on src/server.ts:
```diff
@@ -0,0 +1,64 @@
+const PORT = 5000;
+app.listen(PORT, () => console.log(`🚀 Server running on http://localhost:${PORT}`));
+
+export default app; // Useful for testing
```
useless comment

ClemZi02 commented on src/server.ts:
```diff
@@ -0,0 +1,64 @@
+
+// Start the server on port 5000
+const PORT = 5000;
+app.listen(PORT, () => console.log(`🚀 Server running on http://localhost:${PORT}`));
```
no console.log in code

TOLOES reviewed (changes requested):
you must not create a branch on EPIC, EPIC must not have a product code
